source/common/network/dns_impl.h

```diff
@@ -1,47 +1,68 @@
 #pragma once
 
+#include "envoy/event/file_event.h"
 #include "envoy/network/dns.h"
 
 #include "common/common/linked_object.h"
 #include "common/event/dispatcher_impl.h"
 
+#include "ares.h"
 #include "event2/event_struct.h"
 
 namespace Network {
 
+class DnsResolverImplPeer;
```

Comment:
Generally prefer not giving test code direct access. Why is this necessary? Can we refactor to make it not needed?